### PR TITLE
Add extreme points channel

### PR DESCRIPTION
**Pull request by madil90**
### Description
AddExtremePointsChannel transform adds extreme points from label to the image as new channel. This is used to generate initial segmentation mask for annotation models.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh --codeformat --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick`.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/transforms/utility/array.py [hunk 1]
```diff
@@ -535,3 +536,78 @@ def __call__(
...
 
         return fg_indices, bg_indices
+
+
+class AddExtremePointsChannel(Transform, Randomizable):
+    """
+    Add extreme points of label to the image as a new channel. This transform generates extreme
+    point from label and applies a gaussian filter. The pixel values in points image are rescaled
+    to range [rescale_min, rescale_max] and added as a new channel to input image. The algorithm is
+    described in https://arxiv.org/pdf/2009.11988.pdf.
```
Comment (wyli) [inline monai/transforms/utility/array.py:546]:
```suggestion
    described in Roth et al., Going to Extremes: Weakly Supervised Medical Image Segmentation, https://arxiv.org/abs/2009.11988
```

Comment (madil90) [reply monai/transforms/utility/array.py]: Fixed

File: monai/transforms/utils.py [hunk 2]
```diff
@@ -571,3 +571,56 @@ def get_largest_connected_component_mask(img: torch.Tensor, connectivity: Option
...
+    """
+    Generate extreme points from an image. These are used to generate initial segmentation
+    for annotation models. An optional permutation can be passed to simulate user clicks.
+
+    Args:
+        img:
+            Image to generate extreme points from. Expected Shape is ``(spatial_dim1, [, spatial_dim2, ...])``.
+        rand_state: `np.random.RandomState` object used to select random indices.
+        background: Value to be consider as background, defaults to 0.
+        permutation: Random permutation amount to add to the points, defaults to 0.0.
```
Comment (wyli) [inline monai/transforms/utils.py:588]: do you mean random perturbation? I don't quite understand the permutation here

**Review by wyli — APPROVED** [review]

**Pull request merged**